File: app/src/main/java/org/deiverbum/app/data/entity/PrayerEntity.java
package org.deiverbum.app.data.entity;

import static org.deiverbum.app.utils.Constants.PRAYER;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.deiverbum.app.model.Pater;
import org.deiverbum.app.model.Prayer;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */

@Entity(tableName = PRAYER)
public class PrayerEntity {
    @NonNull
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "prayerID")
    public Integer oracionId=0;

    @NonNull
    @ColumnInfo(name = "prayer")
    public String texto="";

    @NonNull
    @ColumnInfo(name = "order")
    public Integer orden=0;

    @NonNull
    public String getTexto() {
        return texto;
    }

    public Prayer getDomainModel() {
        Prayer dm=new Prayer();
        dm.setPrayer(texto);
        return dm;
    }

}

File: app/src/main/java/org/deiverbum/app/data/entity/PrayerEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.deiverbum.app.model.Prayer
import org.deiverbum.app.util.Constants

/**
 * Entidad para la tabla **`prayer`** de la base de datos, que se ocupa de gestionar las Oraciones.
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(tableName = Constants.PRAYER)
data class PrayerEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "prayerID")
    val oracionId: Int,

    @ColumnInfo(name = "prayer")
    var texto: String,

    @ColumnInfo(name = "order")
    var orden: Int
) {
    val domainModel: Prayer
        get() {
            val dm = Prayer()
            dm.prayer = texto
            return dm
        }
}